File: v-conf-java/v-conf/v-conf/src/main/java/com/example/dto/DefaultConfigurationDTO.java
package com.example.dto;

public class DefaultConfigurationDTO {

    private Integer id;
    private String name;
    private String compName; // Component category: "Air Bags", "Transmission"
    private String compType; // S / I / E / C

    public DefaultConfigurationDTO(
            Integer id,
            String name,
            String compName,
            String compType) {
        this.id = id;
        this.name = name;
        this.compName = compName;
        this.compType = compType;
    }

    public Integer getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getCompName() {
        return compName;
    }

    public String getCompType() {
        return compType;
    }
}

File: v-conf-c#/vehicle_config_c#/project_vc#/project_vc#/DTOs/DefaultConfigurationDTO.cs
namespace project_vc_.DTOs;

public class DefaultConfigurationDTO
{
    public int Id { get; set; }
    public string? Name { get; set; }
    public string? CompName { get; set; }
    public string? CompType { get; set; }

    public DefaultConfigurationDTO() { }
    public DefaultConfigurationDTO(int id, string? name, string? compName, string? compType)
    {
        Id = id;
        Name = name;
        CompName = compName;
        CompType = compType;
    }
}
